aws-logs-destination/src/main/java/software/amazon/logs/destination/BaseHandlerStd.java

```diff
@@ -0,0 +1,115 @@
+            final Logger logger);
+
+    protected CallChain.Completed<DescribeDestinationsRequest, DescribeDestinationsResponse, CloudWatchLogsClient, ResourceModel, CallbackContext> preCreateCheck(
+            final AmazonWebServicesClientProxy proxy, final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient, final ResourceModel model) {
+
+        return proxy.initiate("AWS-Logs-Destination::Create::PreExistenceCheck", proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::translateToReadRequest)
+                .makeServiceCall((awsRequest, sdkProxyClient) -> sdkProxyClient.injectCredentialsAndInvokeV2(awsRequest,
+                        sdkProxyClient.client()::describeDestinations))
+                .handleError((request, exception, client, model1, context1) -> {
+                    ProgressEvent<ResourceModel, CallbackContext> progress;
+                    if (exception instanceof InvalidParameterException) {
+                        progress = ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                                exception.getMessage());
+                    } else if (exception instanceof ServiceUnavailableException) {
+                        progress = ProgressEvent.failed(model, callbackContext, HandlerErrorCode.ServiceInternalError,
+                                exception.getMessage());
+                    } else {
+                        progress = ProgressEvent.progress(model, callbackContext);
+                    }
+                    return progress;
+                });
+    }
+
+    protected boolean isDestinationListNullOrEmpty(final DescribeDestinationsResponse response) {
+
+        return ! response.hasDestinations() || response.destinations()
+                .isEmpty();
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> putDestination(final AmazonWebServicesClientProxy proxy,
+            final CallbackContext callbackContext, final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final ResourceModel model, final String callGraph, final Logger logger, Action handlerAction) {
+
+        return proxy.initiate(callGraph, proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::translateToPutDestinationRequest)
+                .makeServiceCall((awsRequest, sdkProxyClient) -> {
+                    PutDestinationResponse putDestinationResponse = null;
+                    try {
+                        putDestinationResponse = proxyClient.injectCredentialsAndInvokeV2(awsRequest,
+                                proxyClient.client()::putDestination);
+                        logger.log(String.format("%s resource with name %s has been successfully %s",
+                                ResourceModel.TYPE_NAME, model.getDestinationName(), handlerAction.name()));
+                    } catch (AwsServiceException e) {
```

Comment:
we dont catch service ones in handlers those are caught by framework. maybe add more client exceptions?

Reply:
The translate method is handling only the exceptions of type CloudWatchLogsException. All the client exceptions extend CloudWatchLogsException. So all client exceptions are handled. To avoid confusion I replaced the catch statement with CloudWatchLogsException instead of AWSServiceException.